pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -773,10 +775,10 @@ func implicitWaypointPolicies(ctx krt.HandlerContext, Waypoints krt.Collection[W
 		return nil
 	}
 	serviceWaypointKeys := slices.MapFilter(services, func(si model.ServiceInfo) *string {
-		if si.Waypoint == "" || (waypoint != nil && waypoint.ResourceName() == si.Waypoint) {
+		if si.Waypoint.ResourceName == "" || (waypoint != nil && waypoint.ResourceName() == si.Waypoint.ResourceName) {
```

Comment:
what would happen if `si.Waypoint` was nil? is it caught before here?

Reply:
Actually, maybe we handle it before in `fetchWaypointForWorkload`?

```go
func fetchWaypointForWorkload(ctx krt.HandlerContext, Waypoints krt.Collection[Waypoint],
	Namespaces krt.Collection[*v1.Namespace], o metav1.ObjectMeta,
) (*Waypoint, *model.StatusMessage) {
	w, err := fetchWaypointForTarget(ctx, Waypoints, Namespaces, o)
	if err != nil || w == nil {
		return nil, err
	}
	if w.TrafficType == constants.WorkloadTraffic || w.TrafficType == constants.AllTraffic {
		return w, nil
	}
	// Waypoint does not support Workload traffic
	log.Debugf("Unable to add waypoint %s/%s; traffic type %s not supported for %s/%s",
		w.Namespace, w.Name, w.TrafficType, o.Namespace, o.Name)
	return nil, ReportWaypointUnsupportedTrafficType(w.ResourceName(), constants.WorkloadTraffic)
}
```

If that's the case, then disregard/resolve this comment and the other one I made in this review regarding checking nil Waypoint... since it looks like the function call before it's use, properly handles nil Waypoint (which, it looks like it does).

Reply:
Waypoint is not a pointer so it cannot be nil